lib/kafka/producer_interceptors.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Kafka
+  # Holds a list of Kafka::ProducerInterceptor
+  # and wraps calls to a chain of custom producer interceptors.
+  class ProducerInterceptors
+    def initialize(interceptors:, logger:)
+      @interceptors = interceptors
+      @logger = TaggedLogger.new(logger)
+    end
+
+    # This method is called when the client produces a message, before it gets queued to be sent to the Kafka brokers.
+    # The message returned from the first ProducerInterceptor.on_send is passed to the second interceptor on_send, and so on in an
+    # interceptor chain. This method does not throw exceptions.
+    #
+    # @param pending_message [Kafka::PendingMessage] the produced message.
+    #
+    # @return [Kafka::PendingMessagePendingMessage] the intercepted message returned by the last interceptor.
```

Comment:
It doesn't look like the `ConsumerInterceptors` have an equivalent amount of documentation.

Reply:
added more docs